code/game/machinery/computer/crew.dm

```diff
@@ -21,64 +25,78 @@
-	var/list/jobs
+
+	/// List of user -> UI source
+	var/list/ui_sources = list()
+
+	/// Cache of data generated by z-level, used for serving the data within SENSOR_UPDATE_PERIOD of the last update
 	var/list/data_by_z = list()
+
+	/// Cache of last update time for each z-level
 	var/list/last_update = list()
 
-/datum/crewmonitor/New()
-	. = ..()
-
-	var/list/jobs = new/list()
-	jobs["Captain"] = 00
-	jobs["Head of Personnel"] = 50
-	jobs["Head of Security"] = 10
-	jobs["Warden"] = 11
-	jobs["Security Officer"] = 12
-	jobs["Detective"] = 13
-	jobs["Brig Physician"] = 14
-	jobs["Deputy"] = 16
-	jobs["Chief Medical Officer"] = 20
-	jobs["Chemist"] = 21
-	jobs["Geneticist"] = 22
-	jobs["Virologist"] = 23
-	jobs["Medical Doctor"] = 24
-	jobs["Paramedic"] = 25
-	jobs["Psychiatrist"] = 26 //psychiatrist will count as medical, even being gimmick
-	jobs["Research Director"] = 30
-	jobs["Scientist"] = 31
-	jobs["Roboticist"] = 32
-	jobs["Exploration Crew"] = 33
-	jobs["Chief Engineer"] = 40
-	jobs["Station Engineer"] = 41
-	jobs["Atmospheric Technician"] = 42
-	jobs["Quartermaster"] = 51
-	jobs["Shaft Miner"] = 52
-	jobs["Cargo Technician"] = 53
-	jobs["Bartender"] = 61
-	jobs["Cook"] = 62
-	jobs["Botanist"] = 63
-	jobs["Curator"] = 64
-	jobs["Chaplain"] = 65
-	jobs["Clown"] = 66
-	jobs["Mime"] = 67
-	jobs["Janitor"] = 68
-	jobs["Lawyer"] = 69
-	jobs["Barber"] = 70
-	jobs["Stage Magician"] = 71
-	jobs["VIP"] = 72
-	jobs["Admiral"] = 200
-	jobs["CentCom Commander"] = 210
-	jobs["Custodian"] = 211
-	jobs["Medical Officer"] = 212
-	jobs["Research Officer"] = 213
-	jobs["Emergency Response Team Commander"] = 220
-	jobs["Security Response Officer"] = 221
-	jobs["Engineer Response Officer"] = 222
-	jobs["Medical Response Officer"] = 223
-	jobs["Assistant"] = 999 //Unknowns/custom jobs should appear after civilians, and before assistants
-
-	src.jobs = jobs
+	/// Map of job to ID for sorting purposes
+	var/list/jobs = list(
+		// Note that jobs divisible by 10 are considered heads of staff, and bolded
+		// 00: Captain
+		"Captain" = 00,
+		// 10-19: Security
+		"Head of Security" = 10,
+		"Warden" = 11,
+		"Security Officer" = 12,
+		"Detective" = 13,
+		"Brig Physician" = 14,
+		"Deputy" = 15,
+		// 20-29: Medbay
+		"Chief Medical Officer" = 20,
+		"Chemist" = 21,
+		"Geneticist" = 22,
+		"Virologist" = 23,
+		"Medical Doctor" = 24,
+		"Paramedic" = 25,
+		"Psychiatrist" = 26,
+		// 30-39: Science
+		"Research Director" = 30,
+		"Scientist" = 31,
+		"Roboticist" = 32,
+		"Exploration Crew" = 33,
+		// 40-49: Engineering
+		"Chief Engineer" = 40,
+		"Station Engineer" = 41,
+		"Atmospheric Technician" = 42,
+		// 50-59: Cargo
+		"Head of Personnel" = 50,
+		"Quartermaster" = 51,
+		"Shaft Miner" = 52,
+		"Cargo Technician" = 53,
+		// 60+: Civilian/other
+		"Bartender" = 61,
+		"Cook" = 62,
+		"Botanist" = 63,
+		"Curator" = 64,
+		"Chaplain" = 65,
+		"Clown" = 66,
+		"Mime" = 67,
+		"Janitor" = 68,
+		"Lawyer" = 69,
+		"Barber" = 70,
```

Comment:
Won't barbers be considered a head of staff because it's divisible by 10?

Reply:
You are correct, they will. Thanks for pointing that out!